exercises/concept/valentines-day/src/ValentinesDay.hs

```diff
@@ -0,0 +1,21 @@
+module ValentinesDay where
+
+-- Define the remaining data types to implement rateActivity.
+-- You are provided the resulting type (`Approval`) to get you started.
+-- You have to implement the other types (Cuisine, Genre, and Activity),
+-- but they do not need to `derive` from type classes.
+
+data Approval
+  = Yes
+  | No
+  | Maybe
+  deriving (Show, Eq)
```

Comment:
Why provide this data type? I'd prefer the student to define all `data` types. Secondly, can be somehow ensure that students don't need to use `deriving`?

Reply:
@MatthijsBlom / @petertseng do you know a way to test type equality (e.g. `rateActivity Chill 'shouldBe' No`) without deriving `Eq`?

Reply:
@ErikSchierboom 
I have been researching this and can't find a way (I'm pretty sure it is not possible) to test the type constructor without having the type derive `Eq` and `Show`. If that is the case, providing the type as a starting point with a comment indicating it is not representative of the types that need to be implemented is the most straightforward (for the student) way to present this IMO. What do you think?

Reply:
I imagine it would be possible to not have `deriving` if using something other than `shouldBe` in the tests. For example, perhaps you'd use `shouldSatisfy`. But, the tests would be significantly harder to understand. You could give it a try to see how it looks but I wouldn't get my hopes up

Reply:
I worked around it by using `case` expressions in the tests. The tests are a bit harder to read, but definitely simplifies the exercise for the student.

Reply:
Lovely @pwadsworth !